api/envoy/api/v2/listener/listener.proto

```diff
@@ -121,6 +121,16 @@ message FilterChainMatch {
   // * ``tls`` - set by :ref:`envoy.listener.tls_inspector <config_listener_filters_tls_inspector>`
   //   when TLS protocol is detected.
   string transport_protocol = 9;
+
+  // If non-empty, a list of application protocols to consider when determining a filter chain
+  // match. Those values will be compared against the application protocols of a new connection,
+  // when detected by one of the listener filters (this needs to be configured separately).
```

Comment:
Will setting this cause automatic insertion of tls_inspector? If so, I think this comment is incorrect. If not, that's a strange discrepancy from the sni filtering.

Reply:
Oh, good catch! It doesn't right now, and I completely missed it.

Reply:
Erm... but with the TLS Inspector being automatically injected (already fixed), how is this comment incorrect? Do you mean the `(this needs to be configured separately)` part? If so, it's also in the comment for the `transport_protocol` above. I didn't remove it while adding automatic injection, since we prefer users to configure it explicitly and emit a warning when it's not... Should I remove it?

Reply:
Yes, I was referring to `(this needs to be configured separately)`. I'd say remove it from both places.

Reply:
TLS Inspector automatically added, and comments removed. Thanks for catching this!